src/parsec/agent/runners/lua/impl.cpp

```diff
@@ -299,4 +300,28 @@ namespace cbdc::parsec::agent::runner {
 
         return 0;
     }
+	
+    auto lua_runner::make_hash(lua_State* L) -> int {
+        int n = lua_gettop(L);
+        if(n != 1) {
+            lua_pushliteral(L, "hash takes 1 arg");
+            lua_error(L);
+        }
+        if(lua_isstring(L, 1) != 1) {
+                lua_pushliteral(L, "invalid preimage");
+                lua_error(L);
+            }
+        size_t sz{};
+        const auto* str = lua_tolstring(L, 1, &sz);
+        assert(str != nullptr);
+        auto sha = CSHA256();
+        auto unsigned_str = std::vector<unsigned char>(sz);
+        std::memcpy(unsigned_str.data(), str, sz);
+        sha.Write(unsigned_str.data(), sz);
+        cbdc::hash_t computed_hash{};
+        sha.Finalize(computed_hash.data());
```

Comment:
`cbdc::hash_data()` is effectively this snippet refactored into a method; I might suggest using it here directly.